### ISSUE TITLE
Physical Adversarial Examples Against Deep Neural Networks  The Berkeley Artificial Intelligence Research Blog

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#1192]
Physical Adversarial Examples Against Deep Neural Networks – The Berkeley Artificial Intelligence Research Blog<br>
<br>
https://ift.tt/2lrPZMX<br>
<br>
Daniel Seita<br>
<br>

<div>
<div>
  <h1>Physical Adversarial Examples Against Deep Neural Networks</h1>
  <p>
  
  Ivan Evtimov, Kevin Eykholt, Earlence Fernandes, and Bo Li   
  
  
  Dec 30, 2017
  
  </p>
</div>

<article>
  <p><em>This post is based on recent research by Ivan Evtimov, Kevin Eykholt, Earlence
Fernandes, Tadayoshi Kohno, Bo Li, Atul Prakash, Amir Rahmati, Dawn Song, and
Florian Tramèr.</em></p>

<p>Deep neural networks (DNNs) have enabled great progress in a variety of
application areas, including image processing, text analysis, and speech
recognition. DNNs are also being incorporated as an important component in many
cyber-physical systems. For instance, the vision system of a self-driving car
can take advantage of DNNs to better recognize pedestrians, vehicles, and road
signs. However, recent research has shown that DNNs are vulnerable to
<em>adversarial examples</em>: Adding carefully crafted adversarial perturbations to the
inputs can mislead the target DNN into mislabeling them during run time. Such
adversarial examples raise security and safety concerns when applying DNNs in
the real world. For example, adversarially perturbed inputs could mislead the
perceptual systems of an autonomous vehicle into misclassifying road signs, with
potentially catastrophic consequences.</p>

<p>There have been several techniques proposed to generate <em>adversarial examples</em>
and to defend against them. In this blog post we will briefly introduce
state-of-the-art algorithms to generate digital adversarial examples, and
discuss our algorithm to generate <strong>physical</strong> adversarial examples on real
objects under varying environmental conditions. We will also provide an update
on our efforts to generate physical adversarial examples for object detectors.</p>

<h1 id="digital-adversarial-examples">Digital Adversarial Examples</h1>

<p>Different methods have been proposed to generate adversarial examples in the
white-box setting, where the adversary has full access to the DNN. The white-box
setting assumes a powerful adversary and thus can help set the foundation for
developing future fool-proof defenses. These methods contribute to understanding
digital adversarial examples.</p>

<p>Goodfellow et al. proposed the <a href="https://arxiv.org/abs/1412.6572">fast gradient method</a> that applies a
first-order approximation of the loss function to construct adversarial samples.</p>

<p><a href="https://arxiv.org/abs/1608.04644">Optimization</a> based methods have also been proposed to create adversarial
perturbations for targeted attacks.  Specifically, these attacks formulate an
objective function whose solution seeks to maximize the difference between the
true labeling of an input, and the attacker’s desired target labeling, while
minimizing how different the inputs are, for some definition of input
similarity.  In computer vision classification problems, a common measure is the
L2-norm of the input vectors. Often, inputs with low L2 distances will be closer
to each other. Thus, it is possible to compute inputs that are visually very
similar to the human eye, but to a classifier, are very different.</p>

<p>Recent work has examined the <a href="https://arxiv.org/abs/1605.07277">black-box</a> transferability of digital adversarial
examples, generating adversarial examples in black-box settings is also
possible. These techniques involve generating adversarial examples for another
known model in a white-box manner, and then running them against the target
unknown model.</p>

<h1 id="physical-adversarial-examples">Physical Adversarial Examples</h1>

<p>To better understand these vulnerabilities, there has been extensive research on
how <em>adversarial examples</em> may affect DNNs deployed in the <em>physical world</em>.</p>

<p><a href="https://arxiv.org/abs/1607.02533">Kurakin et al.</a> showed that printed adversarial examples can be
misclassified when viewed through a smartphone camera. <a href="https://www.cs.cmu.edu/~sbhagava/papers/face-rec-ccs16.pdf">Sharif et al.</a>
attacked face recognition systems by printing adversarial perturbations on the
frames of eyeglasses. Their work demonstrated successful physical attacks in
relatively stable physical conditions with little variation in pose,
distance/angle from the camera, and lighting. This contributes an interesting
understanding of physical examples in stable environments.</p>

<p>Our recent work “<a href="https://arxiv.org/abs/1707.08945">Robust physical-world attacks on deep learning models</a>” has
shown physical attacks on <strong>classifiers</strong>. (Check out the <a href="https://www.youtube.com/watch?v=1mJMPqi2bSQ&amp;feature=youtu.be">videos</a>
<a href="https://www.youtube.com/watch?v=xwKpX-5Q98o&amp;feature=youtu.be">here</a>.) Concurrent to our work, <a href="https://arxiv.org/pdf/1707.07397.pdf">Athalye et al</a> showed that digitally
optimizing over the transformations, which adversarial inputs would undergo in
the physical world, can yield effective adversarial examples for 3D Objects
against classifiers. As the next logical step, we show attacks on object
<strong>detectors</strong>.  These computer vision algorithms identify relevant objects in a
scene and predict bounding boxes indicating objects’ position and kind. Compared
with classifiers, detectors are more challenging to fool as they process the
entire image and can use contextual information (e.g. the orientation and
position of the target object in the scene) in their predictions.</p>

<p>We demonstrate <em>physical</em> adversarial examples against the <a href="https://pjreddie.com/darknet/yolo/">YOLO</a> detector, a
popular state-of-the-art algorithm with good real-time performance. Our examples
take the form of sticker perturbations that we apply to a real STOP sign. The
following image shows our example physical adversarial perturbation.</p>

<p>
<img src="http://bair.berkeley.edu/static/blog/yolo/image1.png">
<img src="http://bair.berkeley.edu/static/blog/yolo/image4.png">
<br>
</p>

<p>We also perform dynamic tests by recording a video to test out the detection
performance.  As can be seen in the video, the YOLO network does not perceive
the STOP sign in almost all the frames. If a real autonomous vehicle were
driving down the road with such an adversarial STOP sign, it would not see the
STOP, possibly leading to a crash at an intersection. The perturbation we
created is robust to changing distances and angles – the most commonly changing
factors in a self-driving scenario.</p>

<p>More interestingly, the physical adversarial examples generated for the YOLO
detector are also be able to fool standard <a href="https://github.com/endernewton/tf-faster-rcnn">Faster-RCNN</a>. Our demo videos
contains a dynamic test of the physical adversarial example on Faster-RCNN. As
this is a black box attack on Faster-RCNN, the attack is not as successful as it
is in the YOLO case. This is expected behavior. We believe that with additional
techniques (such as ensemble training), the black box attack could be made more
effective.  Additionally, specially optimizing an attack for Faster-RCNN will
yield better results. We are currently working on a paper that explores these
attacks in more detail. The image below is an example of Faster-RCNN not
perceiving the STOP sign.</p>

<p>
<img src="http://bair.berkeley.edu/static/blog/yolo/image3.png">
<img src="http://bair.berkeley.edu/static/blog/yolo/image2.png">
<br>
</p>

<p>In both cases (YOLO and Faster-RCNN), a STOP sign is detected only when the
camera is very close to the sign (about 3 to 4 feet away). In real settings,
this distance is too close for a vehicle to take effective corrective action.
Stay tuned for our upcoming paper that contains more details about the algorithm
and results of physical perturbations against state-of-the-art object detectors.</p>

<h1 id="attack-algorithm-overview">Attack Algorithm Overview</h1>

<p>This algorithm is based off our earlier work on attacking classifiers.
Fundamentally, we take an optimization approach to generating adversarial
examples. However, our experimental experience indicates that generating robust
physical adversarial examples for detectors requires simulating a larger set of
varying physical conditions than what is needed to fool classifiers. This is
likely because a detector takes much more contextual information into account
while generating predictions. Key properties of the algorithm include the
ability to specify sequences of physical condition simulations, and the ability
to specify the translation invariance property. That is, a perturbation should
be effective no matter where the target object is situated within the scene. As
an object can move around freely in the scene depending on the viewer,
perturbations not optimized for this property will likely break when the object
moves. Our upcoming paper on this topic will contain more details on this
algorithm.</p>

<h1 id="potential-defenses">Potential defenses</h1>

<p>Given these adversarial examples in both digital and physical world, potential
defense methods have also been widely studied. Among them, different types of
adversarial training methods are the most effective. <a href="https://arxiv.org/abs/1412.6572">Goodfellow et al.</a>
first proposed adversarial training as an effective way to improve the
robustness of DNNs, and <a href="https://arxiv.org/abs/1705.07204">Tramèr et al.</a> extend it to ensemble adversarial
learning.  <a href="https://pdfs.semanticscholar.org/bcf1/1c7b9f4e155c0437958332507b0eaa44a12a.pdf">Madry et al.</a> have also proposed robust networks via iterative
training with adversarial examples. To conduct an adversarial training based
defense, a large number of adversarial examples are required. In addition, these
adversarial examples can make the defense more robust if they come from
different models as suggested by work on <a href="https://arxiv.org/abs/1705.07204">ensemble training</a>. The benefit of
ensemble adversarial training is to increase the diversity of adversarial
examples so that the model can fully explore the adversarial example space.
There are other types of defense methods as well, but <a href="http://nicholas.carlini.com/papers/2017_aisec_breakingdetection.pdf">Carlini and Wagner</a>
have shown that none of these existing defense method is robust enough given
adaptive attack.</p>

<p>Overall, we are still a long way from finding the optimal defense strategy
against these adversarial examples, and we are looking forward to exploring this
exciting research area.</p>

<h4> Physical Adversarial Sticker Perturbations for YOLO </h4>
VIDEO

<h4> Physical Adversarial Examples for YOLO (2) </h4>
VIDEO

<h4> Black box transfer to Faster RCNN of physical adversarial examples generated for YOLO </h4>
VIDEO

</article>

</div><br>
<br>
<br>
<br>
<br>
<br>
via The Berkeley Artificial Intelligence Research Blog https://ift.tt/2EPVVfk<br>
<br>
October 24, 2018 at 11:52AM
